### ISSUE TITLE
Frozen Pydantic model is not hashable

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#2784]
**Describe the bug**
Pydantic 1.9 implements the dataclass transform spec for its `BaseModel` class.  Unlike dataclasses, when a Pydantic model is frozen, Pyright does not consider it to be hashable.

**To Reproduce**
Install Pydantic and type check the snippet below.  To install Pydantic from source, run `python -m pip install --use-pep517 'pydantic @ git+https://github.com/samuelcolvin/pydantic@v1.9.0'`.

**Expected behavior**
That `Foo()` would be hashable.

**Screenshots or Code**
```python
# pyright: strict

from typing import Hashable

from pydantic import BaseModel

class Foo(BaseModel, frozen=True):
    pass

foo: Hashable = Foo()  # Error: Expression of type "Foo" cannot be assigned to declared type "Hashable"
reveal_type(Foo.__hash__)  # Type of "Foo.__hash__" is "None"


from dataclasses import dataclass

@dataclass(frozen=True)
class Bar:
    pass

bar: Hashable = Bar()  # No error
reveal_type(Bar.__hash__)  # Type of "Bar.__hash__" is "(self: Bar) -> int"

```

**VS Code extension or command-line**
Extension, v1.1.202

(HNY :)

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.203, which I just published. It will also be included in the next release of pylance.
